### PR TITLE
[PARTI-313] Salesforce: Allow defining the ingestion strategy on the cartridge jobs

### PR DESCRIPTION
**Pull request by guiquintelas**
# PR Description

This PR adds the following changes:

- Updates `@constructor-io/constructorio-node` to version `4.4.7`
- Exposes new ingestion `PATCH` types 

## ⚠️ Dependencies

- None

## Ingestion data, with a missing item

```ts
await ingestor.ingest(async () => {
    return {
      data: {
        // insert your test data here
        groups: [],
        items: [
          {
            id: "non-existing-id",
            item_name: "non-existing-item",
          },
        ],
        variations: [],
      },
      type: CatalogIngestionType.PATCH_DELTA_IGNORE,
    };
```

## Running the ingestion
image

## Ingestion result
image

# How to test

## 🧪 What should be tested?

As a reviewer, you should test:

- All new patch ingestions work
- Other ingestion types remain working

## 📝 How to test?

Reproduce the following steps to test:

1. Update the `dev.ts` file with test cases
2. Run `npm run dev`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,17 +104,28 @@ catalogIngestor.ingest(() => ({
...
 import { CatalogIngestionType } from "@constructor-io/catalog-ingestor";
 
 catalogIngestor.ingest(() => ({
-  type: CatalogIngestionType.PATCH,
+  type: CatalogIngestionType.PATCH_DELTA_FAIL,
   // ...other options
 }));
 ```
 
+There are three different patch ingestion types, each one handles missing catalog entities with specific rules:
```
Comment (pedro-lb) [inline README.md:118]: Minor doc thingy

```suggestion
There are three different patch delta ingestion types, and each one handles missing catalog entities with specific rules:
```

File: README.md [hunk 2]
```diff
@@ -104,17 +104,28 @@ catalogIngestor.ingest(() => ({
 
 ### Patch ingestions
```
Comment (pedro-lb) [inline README.md:105]: Same as other discussion

```suggestion
### Patch delta ingestions
```

File: README.md [hunk 3]
```diff
@@ -104,17 +104,28 @@ catalogIngestor.ingest(() => ({
...
 
-**Patch ingestions** update the catalog with the data provided. This strictly updates already existing data. Different from the other two types, it support ingesting partial data.
+**Patch ingestions** update the catalog with the data provided. This strictly updates already existing data. Different from the other two types, it support ingesting partial data.
```
Comment (pedro-lb) [inline README.md:107]: Same

```suggestion
**Patch delta ingestions** update the catalog with the data provided. This strictly updates already existing data. Different from the other two types, it support ingesting partial data. 
```

File: README.md [hunk 4]
```diff
@@ -104,17 +104,28 @@ catalogIngestor.ingest(() => ({
...
+  type: CatalogIngestionType.PATCH_DELTA_FAIL,
   // ...other options
 }));
 ```
 
+There are three different patch ingestion types, each one handles missing catalog entities with specific rules:
+
+*A missing catalog entity is one that is referenced by the ingestion but is not found on the catalog.*
+
+| Ingestion Type | If there are any missing entity |
```
Comment (pedro-lb) [inline README.md:122]: Small typo

```suggestion
| Ingestion Type | If there are any missing entities | 
```

File: package.json [hunk 5]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@constructor-io/catalog-ingestor",
-  "version": "1.2.0",
+  "version": "1.2.1",
```
Comment (pedro-lb) [inline package.json:3]: Should we do `1.3.0`? 

Since that will be a new feature in our connectors, would be great to determine this

**Review by pedro-lb — APPROVED** [review]
This is great!! 🔥 

Thanks for the work here @guiquintelas. I've left a few discussions but none are blocking.

We can merge once everything's been taken care of 🚀

File: package.json (hunk 5 above)
Comment (guiquintelas) [reply package.json]: agreed! 
done 00d3bacc4093583851a3ce11546b5d39e1bff39f ✅

**Review by pedro-lb — APPROVED** [review]
🔥

**Pull request merged**